test/e2e/apimachinery/health_handlers.go

```diff
@@ -0,0 +1,138 @@
+[+]poststarthook/start-kube-aggregator-informers ok
+[+]poststarthook/apiservice-registration-controller ok
+[+]poststarthook/apiservice-status-available-controller ok
+[+]poststarthook/kube-apiserver-autoregistration ok
+[+]autoregister-completion ok
+[+]poststarthook/apiservice-openapi-controller ok
+healthz check passed
+`
+	expectedLivezVerbose = `[+]ping ok
+[+]log ok
+[+]etcd ok
+[+]poststarthook/start-kube-apiserver-admission-initializer ok
+[+]poststarthook/generic-apiserver-start-informers ok
+[+]poststarthook/start-apiextensions-informers ok
+[+]poststarthook/start-apiextensions-controllers ok
+[+]poststarthook/crd-informer-synced ok
+[+]poststarthook/bootstrap-controller ok
+[+]poststarthook/rbac/bootstrap-roles ok
+[+]poststarthook/scheduling/bootstrap-system-priority-classes ok
+[+]poststarthook/start-cluster-authentication-info-controller ok
+[+]poststarthook/aggregator-reload-proxy-client-cert ok
+[+]poststarthook/start-kube-aggregator-informers ok
+[+]poststarthook/apiservice-registration-controller ok
+[+]poststarthook/apiservice-status-available-controller ok
+[+]poststarthook/kube-apiserver-autoregistration ok
+[+]autoregister-completion ok
+[+]poststarthook/apiservice-openapi-controller ok
+healthz check passed
+`
+	expectedReadyzVerbose = `[+]ping ok
+[+]log ok
+[+]etcd ok
+[+]informer-sync ok
+[+]poststarthook/start-kube-apiserver-admission-initializer ok
+[+]poststarthook/generic-apiserver-start-informers ok
+[+]poststarthook/start-apiextensions-informers ok
+[+]poststarthook/start-apiextensions-controllers ok
+[+]poststarthook/crd-informer-synced ok
+[+]poststarthook/bootstrap-controller ok
+[+]poststarthook/rbac/bootstrap-roles ok
+[+]poststarthook/scheduling/bootstrap-system-priority-classes ok
+[+]poststarthook/start-cluster-authentication-info-controller ok
+[+]poststarthook/aggregator-reload-proxy-client-cert ok
+[+]poststarthook/start-kube-aggregator-informers ok
+[+]poststarthook/apiservice-registration-controller ok
+[+]poststarthook/apiservice-status-available-controller ok
+[+]poststarthook/kube-apiserver-autoregistration ok
+[+]autoregister-completion ok
+[+]poststarthook/apiservice-openapi-controller ok
+[+]shutdown ok
+healthz check passed
+`
+)
+
+func testPath(client clientset.Interface, path, expectedOutput string) error {
+	var result restclient.Result
+	err := wait.Poll(100*time.Millisecond, 30*time.Second, func() (bool, error) {
+		result = client.CoreV1().RESTClient().Get().RequestURI(path).Do(context.TODO())
+		status := 0
+		result.StatusCode(&status)
+		return status == 200, nil
+	})
+	if err != nil {
+		return err
+	}
+	body, err := result.Raw()
+	if err != nil {
+		return err
+	}
+	if output := string(body); output != expectedOutput {
+		return fmt.Errorf("unexpected result: %s, expected %s", output, expectedOutput)
+	}
+	return nil
+}
+
+var _ = SIGDescribe("health handlers", func() {
+	f := framework.NewDefaultFramework("health")
+
+	ginkgo.It("/healthz should contain all checks", func() {
```

Comment:
three distinct tests seems like overkill... maybe one with three calls it verifies? remember there is per test overhead like namespace creation, etc

Reply:
Done.